app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt

```diff
@@ -69,51 +71,56 @@ class SectionHolder(
                     )
                 }
             } else if (viewModel.errors() != null && viewModel.warnings() != null) {
-                sectionFieldsErrorInfo.visibility = View.GONE
-                sectionFieldsWarningErrorInfo.visibility = View.VISIBLE
-                sectionFieldsWarningInfo.visibility = View.VISIBLE
-                sectionFieldsInfo.visibility = View.GONE
+                sectionFieldsErrorInfo.visibility = GONE
+                sectionFieldsWarningErrorInfo.visibility = VISIBLE
+                sectionFieldsWarningInfo.visibility = VISIBLE
+                sectionFieldsInfo.visibility = GONE
                 sectionFieldsWarningErrorInfo.chipText = viewModel.errors().toString()
                 sectionFieldsWarningInfo.chipText = viewModel.warnings().toString()
             } else if (viewModel.errors() != null && viewModel.warnings() == null) {
-                sectionFieldsErrorInfo.visibility = View.VISIBLE
-                sectionFieldsWarningErrorInfo.visibility = View.GONE
-                sectionFieldsWarningInfo.visibility = View.GONE
-                sectionFieldsInfo.visibility = View.GONE
+                sectionFieldsErrorInfo.visibility = VISIBLE
+                sectionFieldsWarningErrorInfo.visibility = GONE
+                sectionFieldsWarningInfo.visibility = GONE
+                sectionFieldsInfo.visibility = GONE
                 sectionFieldsErrorInfo.chipText = viewModel.errors().toString()
             } else {
-                sectionFieldsErrorInfo.visibility = View.GONE
-                sectionFieldsWarningErrorInfo.visibility = View.GONE
-                sectionFieldsWarningInfo.visibility = View.VISIBLE
-                sectionFieldsInfo.visibility = View.GONE
+                sectionFieldsErrorInfo.visibility = GONE
+                sectionFieldsWarningErrorInfo.visibility = GONE
+                sectionFieldsWarningInfo.visibility = VISIBLE
+                sectionFieldsInfo.visibility = GONE
                 sectionFieldsWarningInfo.chipText = viewModel.warnings().toString()
             }
         }
 
-        formBinding.descriptionIcon.visibility = if (viewModel.description().isNullOrEmpty()) {
-            View.GONE
-        } else {
-            View.VISIBLE
+        viewModel.description().takeIf { it.isNullOrEmpty() }?.let {
+            formBinding.descriptionIcon.visibility = GONE
+            formBinding.sectionName.viewTreeObserver.addOnGlobalLayoutListener {
+                formBinding.sectionName.layout?.let {
+                    if (it.getEllipsisCount(0) > 0) {
+                        formBinding.descriptionIcon.visibility = VISIBLE
+                    }
```

Comment:
Do you think it makes sense to extract this presentation logic? This is a very small example but when it gets bigger we cannot unit test :(

Reply:
yes, you are right, I will extract it. Thanks!.